superset-frontend/src/explore/components/ControlRow.tsx

```diff
@@ -16,17 +16,19 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-import { ExpandedControlItem } from '@superset-ui/chart-controls';
 import React from 'react';
 
 const NUM_COLUMNS = 12;
 
-export default function ControlRow({
-  controls,
-}: {
-  controls: ExpandedControlItem[];
-}) {
-  const colSize = NUM_COLUMNS / controls.length;
+type Control = React.ReactElement | null;
+
+export default function ControlRow({ controls }: { controls: Control[] }) {
+  // ColorMapControl renders null and should not be counted
+  // in the columns number
+  const countableControls = controls.filter(
+    control => !['ColorMapControl'].includes(control?.props.type),
```

Comment:
Specific check required as this needs to be rendered but the component itself is null (within)/invisible to the end-user.